sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -223,8 +229,9 @@ object ScalaNativePluginInternal {
       val opaths = (nativelib ** "*.o").get.map(abs)
       val paths  = apppaths ++ opaths
       val links = {
-        val os   = Option(sys props "os.name").getOrElse("")
-        val arch = compileTarget.split("-").head
+        val os    = Option(sys props "os.name").getOrElse("")
+        val arch  = compileTarget.split("-").head
+        val regex = Seq("re2")
```

Comment:
Use `@link("re2")` on the object with bindings instead. There is no need to link with regex if the bindings are not used.